### ISSUE TITLE
Lots of file descriptors open while testing installed tools

### ISSUE THREAD

**Issue by HugoJH** [galaxyproject/ephemeris#166]
While doing `shed-tools test` over my galaxy instance, had a problems of reaching the limit of maximum number of open files.
For the sake of keep running the test, I modified the limit editing /etc/security/limits. Relaunched the `shed-tools test` command and waited enough time to check the number of open files at that point.

I did `lsof | grep shed-tool | grep -v deleted| wc` and  found out that there where more than 3000 opened files by the shed-tools process. Seems like the files opened for each test  are not gettting closed once the test finishes. There should be a mechanism to close all unnecesary file descriptors once the test finishes.

**Comment by rhpvorderman** [comment]
Hi @HugoJH, I checked out the code and I cannot find the open statement that causes this. 
Can you give us some more information?
- How many tests did you run in parallel? 
- Can you give some of the lsof output? What sort of files does shed-tools open?

**Comment by afgane** [comment]
@almahmoud, you've experienced this behavior as well, right?

**Comment by almahmoud** [comment]
Yes! The problem seems to be in tool-util I think, cause it came up when running the tool tests directly not just with ephemeris. A temporary workaround was setting `ulimit -n 20000` (or some other really high number depending on how many tests are run total)

**Comment by mvdbeek** [comment]
Are these files or sockets that are open ? Can you also check what version of galaxy-tool-util you've got in your environment ? (` pip freeze| grep `galaxy-tool-util`)

**Comment by almahmoud** [comment]
The last time I tried it was installed from source from https://github.com/galaxyproject/galaxy/pull/10767/files

**Comment by mvdbeek** [comment]
Alright, I've traced this too, and https://github.com/galaxyproject/galaxy/pull/10852 prevents accumulating those in a small test.
